validate-data.js

```diff
@@ -0,0 +1,212 @@
+    const isIntermediateMinor = !isFirstMinor && !isFinalMinor;
+
+    // variable which is reused for mutliple assertions
+    let expected;
+
+    // Update globally scope objected
+    version = currObj.version;
+
+    // Validate releaseDate - applies to all scenarios
+    if (hasBeenReleased(currObj)) {
+        scope = 'Released version';
+        if (currObj.releaseDateExtra !== null) {
+            error(`releaseDateExtra must be null`);
+        }
+    } else {
+        scope = 'Unlreleased version';
+        if (!currObj.releaseDate.match(/^(\d{4}-\d{2})$/)) {
+            error(`releaseDate must be in YYYY-MM format`);
+        }
+    }
+
+    // The following has some duplicated logic and extra nesting due to prioritising
+    // readabilitiy over conciseness
+    if (isFirstMinor) {
+        // e.g. 6.0, 7.0
+        scope = 'First minor version';
+        if (currObj.supportLength !== 'Standard') {
+            error(`supportLength must be "Standard"`);
+        }
+        // Standard + Full support - Approximately 6 months, until the next minor release is performed
+        if (nextObj) {
+            // Here we use "until the next minor release is performed"
+            if (currObj.partialSupport !== nextObj.releaseDate) {
+                error(`partialSupport must be "${nextObj.releaseDate}"`);
+            }
+        } else {
+            // Here we use "approximately 6 months", because there is no known next minor release date
+            expected = addTime(currObj.releaseDate, 6, 'months', false);
+            if (currObj.partialSupport !== expected) {
+                error(`partialSupport must be "${expected}"`);
+            }
+        }
+        if (!hasBeenReleased(currObj)) {
+            // This will be an unreleased next major, it needs an April-June release date
+            scope = 'Unreleased first minor version';
+            if (currObj.releaseDateExtra === null) {
+                error(`releaseDateExtra must not be null`);
+            }
+        }
+        // Standard + Partial support - exactly 6 months (+1 day)
+        expected = addTime(currObj.partialSupport, 6, 'months', true);
+        if (currObj.supportEnds !== expected) {
+            error(`supportEnds must be "${expected}"`);
+        }
+
+    } else if (isIntermediateMinor) {
+        // e.g. 5.3, 6.1
+        scope = 'Intermediate minor version';
+        if (currObj.supportLength !== 'Standard') {
+            error('supportLength must be "Standard"');
+        }
+        // Standard + Full support - until the next minor release is performed
+        if (currObj.partialSupport !== nextObj.releaseDate) {
+            error(`partialSupport must be ${nextObj.releaseDate}`);
+        }
+        // Standard + Partial support - exactly 6 months (+1 day)
+        expected = addTime(currObj.partialSupport, 6, 'months', true);
+        if (currObj.supportEnds !== expected) {
+            error(`supportEnds must be "${expected}"`);
+        }
```

Comment:
These all apply for both intermediate _and_ first minor, right?
This duplication honestly makes it a bit harder for me to follow since I have to keep cross-referencing against the duplications to check if they're doing the same thing (and if they're _meant_ to be doing the same thing).

Might be easier to remove the `else` part, and have:

`if (isFirstMinor)` - ONLY things that are for the first minor
`if (!isFinalMinor)` (or `if (!isFinalMinor && !unreleasedMajorSeries)`) - anything that's for BOTH first and intermediate minor. In the second option `unreleasedMajorSeries` is e.g. `7.0` right now. I.e. a future major series which has had no releases so far.
`if (isFinalMinor)` - ONLY things that are for the final minor.

Reply:
Almost but not quite

FirstRelease applies to an unreleased majors such as CMS 7.0, so the logic has to be a bit different for that

The options you listed above are also confusing in their own way, I don't think there's any way to avoid this being a little confusing.

Given that everything works, and we've already spent a fair bit of time on this, I don't think we should refactor this bit

Reply:
I assume by `FirstRelease` you mean `isFirstMinor`? If so, that's why I suggest still having a `isFirstMinor` section first, with anything that only applies to that. 

I'll drop it but if I have to update this logic in the future I may refactor it.